internal/compiler/widgets/material-base/widget-button.slint

```diff
@@ -53,52 +66,33 @@ export Button := Rectangle {
         }
 
         label := Text {
+            text: root.text;
+            font-size: root.font-size;
             color: md.sys.color.on-secondary-container;
             vertical-alignment: center;
             horizontal-alignment: center;
-            font-size: md.sys.typescale.label-large.size;
             font-weight: md.sys.typescale.label-large.weight;
         }
     }
 
-    touch := TouchArea {
-        clicked => {
-            if (root.checkable) {
-                root.checked = !root.checked;
-            }
-            root.clicked();
-        }
-    }
-
-    fs := FocusScope {
-        width: 0px; // Do not react on clicks
-        enabled <=> root.enabled;
-        key-pressed(event) => {
-            if (event.text == " " || event.text == "\n") {
-                 touch.clicked();
-                 return accept;
-            }
-            return reject;
-        }
-    }
-
     states [
         disabled when !root.enabled : {
             container.background: md.sys.color.on-surface;
             container.opacity: 0.12;
             label.opacity: 0.38;
             label.color: md.sys.color.on-surface;
         }
-        pressed when touch.pressed : {
-            state-layer.opacity: 0.12;
+        checked when root.checked: {      
+            label.color: md.sys.color.on-primary;
         }
-        hover when touch.has-hover : {
-            state-layer.opacity: 0.08;    
-            container.drop-shadow-blur: md.sys.elevation.level1;
-            container.drop-shadow-color: md.sys.color.shadow;
+    ]
+
+    transitions [  
```

Comment:
The new transitions way can be used here too.

Reply:
@ogoffart I tried it as it is described in the langref. But I get

```
        in {  
     |                ^
  error: Parse error
```

I also tried it with `:` but it didn't worked ether.

Reply:
where do you get that error?  Maybe you're using an old version of slint?

Reply:
in the widgets-button.slint in my current branch.